MapleServer2/Types/Currency.cs

```diff
@@ -39,6 +42,7 @@ public void SetAmount(long input)
 
         private void UpdateWallet()
         {
+            DatabaseManager.Update(Player.Wallet);
```

Comment:
This feels dicey, it means everytime meso is changed in any way (or any currency), it will make a database call when instead it should be enough to save upon a channel/map change or logging out, and keep the new wallet state in memory